src/jitlibs/softfloat/CMakeLists.txt

```diff
@@ -1,14 +1,405 @@
 PROJECT(softfloat)
 
-# TODO: Add softfloat lib as external project
-ADD_LIBRARY(${PROJECT_NAME}
-    dummy.cpp
+##################### SOFTFLOAT UTILITY PROJECT (berkeley softfloat) #####################
+include(FetchContent)
+FetchContent_Declare(softfloat_srcs
+	GIT_REPOSITORY https://github.com/ucb-bar/berkeley-softfloat-3.git
+	GIT_TAG master
 )
-TARGET_INCLUDE_DIRECTORIES(${PROJECT_NAME} PUBLIC .)
+if(NOT softfloat_srcs_POPULATED)
+	FetchContent_Populate(softfloat_srcs)
+	file(COPY ${softfloat_srcs_SOURCE_DIR}/source/include/softfloat.h DESTINATION ${softfloat_srcs_SOURCE_DIR})
+	file(RENAME ${softfloat_srcs_SOURCE_DIR}/softfloat.h ${softfloat_srcs_SOURCE_DIR}/softfloat_orig.h )
+	FetchContent_GetProperties(softfloat_srcs)
+	set(EXTERNAL_INSTALL_LOCATION ${softfloat_srcs_SOURCE_DIR})
+
+##################### SOFTFLOAT UTILITY PROJECT (berkeley softfloat) #####################
+
+EXECUTE_PROCESS( COMMAND uname -m COMMAND tr -d '\n' OUTPUT_VARIABLE ARCHITECTURE )
+IF(WIN32)
+	IF(NOT CMAKE_CL_64)
```

Comment:
have you tested this? it seems like it wouldn't work, because when using mingw, you are not using msvc cl. The documentation proposes the proper alternative.

Reply:
Only for some Linux machines. 

Are you referring to [4. SoftFloat Package Directory Structure](http://www.jhauser.us/arithmetic/SoftFloat-3/doc/SoftFloat-source.html) as the documentation in question? I can make out the various possible configurations, but I am not sure how to configure this for all possible ETISS use(r) cases.

As far as I understand the ETISS main page, Linux x86-64 and Windows is supported/tested. This means the CMakeLists.txt should be able to detect and configure
- Linux x86-80 w/ 8086-SSE or 8086, or
- Windows and set Win64-MinGW-w64 w/ 8086-SSE or 8086?

Reply:
As far as I understand this page https://cmake.org/cmake/help/latest/variable/CMAKE_CL_64.html , that variable is only set when using cl as compiler, which would not be the case when using mingw. The standard way to check architecture bitness is CMAKE_SIZEOF_VOID_P.

Reply:
Introduced more specific host information retrieval with latest commits [61d12ed](https://github.com/tum-ei-eda/etiss/pull/18/commits/61d12ed7a0c54eb32a9ad8a58e913a2d80560657)

However, build on Windows host still fails with MSVC: 
![grafik](https://user-images.githubusercontent.com/62236820/86154733-c1445b80-bb03-11ea-8fd9-f37b17b8ba97.png)

It seems like, MSVC does not support the '__int128' type introduced through a common header all Windows targets of the source softfloat library. MinGW seems to support this natively.

I am no Windows expert, however, it seems like we will have to add a MSVC-target configuration to the berkeley softfloat library, or replace the header specifiying the '__int128' by a MSVC conform data type.

Right now, this pull request will break Windows support for ETISS.